syntax/table.php

```diff
@@ -114,31 +110,31 @@ public function render($mode, Doku_Renderer $renderer, $data)
         $groups = $data['groups'];
         $attributes = $data['attributes'];
 
-        $users = $auth->retrieveUsers(0,
-            -1,
-            ['grps' => implode('|', $groups)]
-        );
-
-        // no results from auth backend
-        if (empty($users)) {
-            $rows = [];
-        } else {
-            // convert user data into matrix row: attributes and group membership flags
-            $rows = array_map(function ($user, $username) use ($groups, $attributes) {
-                // special handling of 'user': always use the wiki username from array key
-                $user['user'] = $username;
-
-                foreach ($attributes as $attribute) {
-                    $row[$attribute] = $user[$attribute] ?: '';
+        $rows = [];
+        foreach ($groups as $group) {
+            $users = $auth->retrieveUsers(0, -1, ['grps' => $group]);
+            foreach ($users as $user) {
+                if (!isset($rows[$user['user']])) {
+                    // prepare row of attributes + groups
+                    $rows[$user['user']] = array_merge(
```

Comment:
Unfortunately this will not work with authplain, as it does not have the `user` key in user data array. I'm not sure how much sense it would make to replace the lines with something like

```suggestion
            foreach ($users as $key => $user) {
                if (!isset($rows[$user['user']])) {
                $username = isset($user['user']) ? $user['user'] : $key;
                    if (!isset($rows[$username])) {
                        // prepare row of attributes + groups
                        $rows[$username] = array_merge(
```